File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/receive/di/ReceiveComponent.java
package com.github.willjgriff.ethereumwallet.ui.receive.di;

import com.github.willjgriff.ethereumwallet.di.AppComponent;
import com.github.willjgriff.ethereumwallet.di.ControllerScope;
import com.github.willjgriff.ethereumwallet.ui.receive.ReceiveController;

import dagger.Component;

/**
 * Created by Will on 01/03/2017.
 */

@Component(dependencies = AppComponent.class)
@ControllerScope
public interface ReceiveComponent {

	void inject(ReceiveController receiveController);
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/screens/receive/di/ReceiveComponent.kt
package com.github.willjgriff.ethereumwallet.ui.screens.receive.di

import com.github.willjgriff.ethereumwallet.di.AppComponent
import com.github.willjgriff.ethereumwallet.di.AppInjector
import com.github.willjgriff.ethereumwallet.di.ControllerScope
import com.github.willjgriff.ethereumwallet.ui.screens.receive.ReceiveController
import dagger.Component

/**
 * Created by williamgriffiths on 18/04/2017.
 */
@Component(dependencies = arrayOf(AppComponent::class))
@ControllerScope
interface ReceiveComponent {

    fun inject(receiveController: ReceiveController)
}

fun ReceiveController.injectNewReceivePresenter() {
    DaggerReceiveComponent.builder()
            .appComponent(AppInjector.appComponent)
            .build()
            .inject(this)
}